### ISSUE TITLE
upgrading kedro_cli should be more explicit in the 0.15.* to 0.16.* migration guide

### ISSUE THREAD

**Issue by WaylonWalker** [quantumblacklabs/kedro#384]
## Description

I upgraded from a `0.15.9` project, and found that the `kedro_cli` module was quite angry.  Looking back over the migration guide, i found this note in the Migration Guide, within `Migration for kedro env environment variable`.

> Note: If you haven't made significant changes to your kedro_cli.py, it may be easier to simply copy the updated kedro_cli.py .ipython/profile_default/startup/00-kedro-init.py and from GitHub or a newly generated project into your old project.

I completely glossed over this section because I was not utilizing environment variables for this project.

## Simple Resolution

Make this note a bit more apparent inside of `release.md` by giving it its own section.   To be honest I was a bit focused on my task and did not look heavily at the diff, but it was quite different.  It definitely merits its own section of the Guide.

## permanent resolution
_maybe needs another issue created, or might be already under one of the framework issues_

This is a big enough module that it should just sit inside the framework, and simply be imported into kedro_cli, then users can add/overwrite commands inside this file without all of the boiler plate.

## Error

This is the error I got.  Upon first inspection of the final line it doesn't lead me to initially believe that it's due to the upgrade.  I am not sure if its a feasible fix, but a better error message here would be helpful.  I believe the simple resolution highlighted above would be sufficient.

```
  File "/mnt/c/temp/kedro0160/kedro_cli.py", line 49, in <module>
    from kedro.cli.pipeline import pipeline as pipeline_group
ModuleNotFoundError: No module named 'kedro.cli.pipeline'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/me/miniconda3/envs/kedro0160/bin/kedro", line 8, in <module>
    sys.exit(main())
  File "/home/me/miniconda3/envs/kedro0160/lib/python3.8/site-packages/kedro/framework/cli/cli.py", line 623, in main
    _handle_exception(f"Cannot load commands from {kedro_cli_path}")
  File "/home/me/miniconda3/envs/kedro0160/lib/python3.8/site-packages/kedro/framework/cli/cli.py", line 643, in _handle_exception
    raise KedroCliError(msg)
kedro.framework.cli.utils.KedroCliError: Cannot load commands from /mnt/c/temp/kedro0160/kedro_cli.py
```

**Comment by lorenabalan** [comment]
Good spot! I agree that it's not immediately obvious from phrasing the migration guide,  so it'll need a fix there.

>This is a big enough module that it should just sit inside the framework, and simply be imported into kedro_cli, then users can add/overwrite commands inside this file without all of the boiler plate.

Most of this has been addressed in 0.16.*. We made a conscious decision to not remove functionality related to `kedro run`. That commands seemed to be the most likely to be customised, and we wanted to maintain an accessible workflow for beginners to just be able to read/modify the given file, rather than look up the source code, copy-paste various things, etc.
